### PR TITLE
Add convert_finetuning_dataset to CLI

### PR DESCRIPTION
**Pull request by KuuCi**
This PR allows users to call llmfoundry convert_finetuning_dataset {ARGS} while maintaining correctness with existing convert_finetuning_dataset script. The motivation is for DLE where we want to make the CLI much more intuitive in the docker images

### CODE REVIEW

**Comment by KuuCi** [comment]
test-data-txt-cEry3n runs:
`llmfoundry convert_finetuning_dataset \
      --dataset "Muennighoff/P3" \
      --splits train,validation \
      --preprocessor llmfoundry.data.finetuning.tasks:p3_preprocessing_function \
      --out_root data_folder`
      
 test-data-txt-ccNcwV runs:
 `python convert_finetuning_dataset.py \
      --dataset "Muennighoff/P3" \
      --splits train validation \
      --preprocessor llmfoundry.data.finetuning.tasks:p3_preprocessing_function \
      --out_root data_folder`

**Comment by KuuCi** [comment]
ft-cli-JmJNkX runs:
`cd llm-foundry/scripts/data_prep
  mkdir shakespeare && cd shakespeare
  curl -O https://ocw.mit.edu/ans7870/6/6.006/s08/lecturenotes/files/t8.shakespeare.txt
  cd ..
  llmfoundry convert_finetuning_dataset \
    --dataset "Muennighoff/P3" \
    --splits train,validation \
    --preprocessor "llmfoundry.data.finetuning.tasks:p3_preprocessing_function" \
    --out-root "data_folder"`
    
    
    ft-orig-5VBmib runs:
    `cd llm-foundry/scripts/data_prep
  mkdir shakespeare && cd shakespeare
  curl -O https://ocw.mit.edu/ans7870/6/6.006/s08/lecturenotes/files/t8.shakespeare.txt
  cd ..
  python convert_finetuning_dataset.py \
    --dataset "Muennighoff/P3" \
    --splits train validation \
    --preprocessor "llmfoundry.data.finetuning.tasks:p3_preprocessing_function" \
    --out_root "data_folder"`

**Comment by KuuCi** [comment]
mcli logs ft-orig-8sBz3c runs:
`python convert_finetuning_dataset.py \
    --dataset "Muennighoff/P3" \
    --splits train validation \
    --preprocessor "llmfoundry.data.finetuning.tasks:p3_preprocessing_function" \
    --out_root "data_folder"`
    
mcli logs ft-cli-KIX0U6 runs:
`llmfoundry data_prep convert_finetuning_dataset \
    --dataset "Muennighoff/P3" \
    --splits train,validation \
    --preprocessor "llmfoundry.data.finetuning.tasks:p3_preprocessing_function" \
    --out-root "data_folder"`
    
image

File: tests/data/test_dataloader.py [hunk 1]
```diff
@@ -22,6 +22,9 @@
 from streaming.base.util import clean_stale_shared_memory
 
 from llmfoundry.command_utils import convert_dataset_hf
+# yapf: enable
```
Comment (dakinggg) [inline tests/data/test_dataloader.py:25]: remove

**Review by dakinggg — APPROVED** [review]

**Pull request merged**